File: python/src/citronella/page_tab.py
class PageTab:
    """This is a page page tab class."""
    _pages = []

    @property
    def current_page(self):
        """return the last page object stored."""
        return self._pages[-1]

    def append(self, new_page):
        """store the page object in _pages."""
        self._pages.append(new_page)
        if len(self._pages) > 5:
            self._pages.pop(0)

    def pop(self):
        """delete the last item of the _pages lists."""
        if len(self._pages) == 0:
            return
        self._pages.pop()

File: ruby/lib/page_tab.rb
#MIT License
#
#Copyright (c) 2023 Eko Purnomo
#
#Permission is hereby granted, free of charge, to any person obtaining a copy
#of this software and associated documentation files (the "Software"), to deal
#in the Software without restriction, including without limitation the rights
#to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
#copies of the Software, and to permit persons to whom the Software is
#furnished to do so, subject to the following conditions:
#
#The above copyright notice and this permission notice shall be included in all
#copies or substantial portions of the Software.
#
#THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
#IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
#FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
#AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
#LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
#OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
#SOFTWARE.


module Citronella
  class PagesList
    """This is a page page tab class."""
    def initialize
      @pages = []
    end

    def current_page
      """return the last page object stored."""
      @pages.last
    end

    def append(new_page)
      """store the page object in _pages."""
      @pages << new_page
      if @pages.length > 5
        @pages.shift
      end
    end

    def pop
      """delete the last item of the _pages lists."""
      return if @pages.empty?
      @pages.delete_at(-1)
    end
  end
end

